src/components/common/Button.tsx

```diff
@@ -166,7 +157,19 @@ const Button = ({
           <span className='sr-only'>Loading</span>
         </>
       )}
-    </ButtonElement>
+    </>
+  );
+
+  if (typeof buttonAttributes.href === 'string') {
```

Comment:
This is actually the magic part. Before you use the dynamic props and pass it for example to `<Link>` here, which can not accept any button components, you NEED to do the check to tell typescript what props are actually passed in. And since our button component interface has `href?: never`, typescript now knows that the anchor interface HAS to apply here. 
And only after that you can pass all the props in to the component. 

Before you had split this into a separate component, and was only doing the check there. So in this component, TypeScript wouldn't know yet what the type of the button was, and was yelling at you because of that 😁